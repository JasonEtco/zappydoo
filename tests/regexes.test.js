const { regexes } = require('../lib/constants')
const fs = require('fs')
const path = require('path')

describe('regexes', () => {
  let templates
  const pather = str => path.join(__dirname, 'fixtures', 'regexes', str)

  beforeEach(() => {
    templates = {
      blockEnds: fs.readFileSync(pather('block-ends.md'), 'utf8'),
      blockNoEnd: fs.readFileSync(pather('block-no-end.md'), 'utf8')
    }
  })

  describe('block', () => {
    it('returns the correct block', () => {
      const match = regexes.block.exec(templates.blockEnds)
      console.log(match)
    })
  })
})
