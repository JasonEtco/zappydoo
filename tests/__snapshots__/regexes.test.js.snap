// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regexes block returns the correct block 1`] = `
Array [
  "{% name value %}
  I have content!
{% endname %}",
  "name",
  "value",
]
`;

exports[`regexes var returns the correct variable 1`] = `
Array [
  "{{ example }}",
  "example",
]
`;
